### PR TITLE
fix: error "Cannot read properties of undefined (reading 'url')" when mapping metadata.externalReferences

### PR DESCRIPTION
**Pull request by meriouma**
We're now unable to submit our SBOM because of this error
```
Error: Cannot read properties of undefined (reading 'url')
```

### CODE REVIEW

File: __tests__/data/valid-bom-1.4.json [hunk 1]
```diff
@@ -170,7 +174,7 @@
     {
       "ref": "pkg:npm/acme/component@1.0.0",
       "dependsOn": [
-        "pkg:npm/acme/component@1.0.0"
+        "pkg:npm/acme/mylibrary@1.0.0"
```
Comment (meriouma) [inline __tests__/data/valid-bom-1.4.json:177]: This was creating a cycle in method `addIndirectDeps`

**Comment by davidkarlsen** [comment]
@meriouma Was this problem introduced in some recent version of the action or was it always there?

**Comment by meriouma** [comment]
> @meriouma Was this problem introduced in some recent version of the action or was it always there?

Hmm it's hard to tell. It's on a project we didn't not update for a bit. We managed to upload a version with version 0.0.1, 3 months ago, with an older version of cyclonedx-python. Maybe it wasn't adding `externalReferences`. The way the access to `externalReferences` values is coded, I'd believe the issue was always there.

**Review by kjetiloen — APPROVED** [review]
lgtm, thank you!

**Pull request merged**